### PR TITLE
Added function to connect to server without ssl

### PR DESCRIPTION
**Pull request by JLHasson**
Creates a standard method for connecting to servers without SSL certificates or with self-signed certs. This is to avoid having to create custom SSL contexts for every script that wishes to access a server without SSL, and also to avoid having contexts being created incorrectly (i.e. globally).

### CODE REVIEW

File: pyVim/connect.py [hunk 1]
```diff
@@ -804,6 +804,37 @@ def SmartConnect(protocol='https', host='localhost', port=443, user='root', pwd=
                   b64token=b64token,
                   mechanism=mechanism)
 
+def ConnectNoSSL(protocol='https', host='localhost', port=443, user='root', pwd='',
```
Comment (tianhao64) [inline pyVim/connect.py]: Thanks @JLHasson 
Can we change the function name to SmartConnectNoSSL as it's a wrapper of SmartConnect? Do you plan to make the same change for Connect?

Comment (JLHasson) [inline pyVim/connect.py]: I can do that. I will extend it to Connect as well.

File: pyVim/connect.py [hunk 2]
```diff
@@ -804,6 +834,37 @@ def SmartConnect(protocol='https', host='localhost', port=443, user='root', pwd=
                   b64token=b64token,
                   mechanism=mechanism)
 
+def SmartConnectNoSSL(protocol='https', host='localhost', port=443, user='root', pwd='',
+                 service="hostd", path="/sdk",
```
Comment (tianhao64) [inline pyVim/connect.py]: Please fix the format here.

File: pyVim/connect.py [hunk 3]
```diff
@@ -271,6 +271,36 @@ def Connect(host='localhost', port=443, user='root', pwd='',
...
+   SSL context and then connect via the Connect method.
+   """
+
+   if hasattr(ssl, '_create_unverified_context'):
+      sslContext = ssl._create_unverified_context()
+   else:
+      sslContext = None
+
+   return Connect(host=host,
+                  port=portNumber,
```
Comment (tianhao64) [inline pyVim/connect.py]: portNumber -> port

File: pyVim/connect.py [hunk 4]
```diff
@@ -271,6 +271,36 @@ def Connect(host='localhost', port=443, user='root', pwd='',
...
+                  user=user,
+                  pwd=pwd,
+                  service=service,
+                  adapter='SOAP',
```
Comment (tianhao64) [inline pyVim/connect.py]: adapter=adapter

File: pyVim/connect.py [hunk 5]
```diff
@@ -271,6 +271,36 @@ def Connect(host='localhost', port=443, user='root', pwd='',
...
+                  version=supportedVersion,
```
Comment (tianhao64) [inline pyVim/connect.py]: version=version

File: pyVim/connect.py (hunk 3 above)
Comment (JLHasson) [inline pyVim/connect.py]: Ah. Testing on a windows VM but using Git from OSX. Slipped my mind

File: pyVim/connect.py [hunk 6]
```diff
@@ -804,6 +834,37 @@ def SmartConnect(protocol='https', host='localhost', port=443, user='root', pwd=
                   b64token=b64token,
                   mechanism=mechanism)
 
+def SmartConnectNoSSL(protocol='https', host='localhost', port=443, user='root', pwd='',
+                      service="hostd", path="/sdk",
+                      preferredApiVersions=None, keyFile=None, certFile=None,
+                      thumbprint=None, sslContext=None, b64token=None, mechanism='userpass'):
```
Comment (tianhao64) [inline pyVim/connect.py]: I think we should remove sslContext since it is ignored inside the method

File: pyVim/connect.py [hunk 7]
```diff
@@ -271,6 +271,36 @@ def Connect(host='localhost', port=443, user='root', pwd='',
 
    return si
 
+def ConnectNoSSL(host='localhost', port=443, user='root', pwd='',
+                 service="hostd", adapter="SOAP", namespace=None, path="/sdk",
+                 version=None, keyFile=None, certFile=None, thumbprint=None,
+                 sslContext=None, b64token=None, mechanism='userpass'):
```
Comment (tianhao64) [inline pyVim/connect.py]: remove sslContext parameter

File: pyVim/connect.py (hunk 6 above)
Comment (JLHasson) [inline pyVim/connect.py]: I wondered about this but wanted to maintain consistency with the main Connect functions

File: pyVim/connect.py [hunk 8]
```diff
@@ -271,6 +271,36 @@ def Connect(host='localhost', port=443, user='root', pwd='',
...
+   or when you wish to ignore SSL altogether. Will attempt to create an unverified
+   SSL context and then connect via the Connect method.
+   """
+
+   if hasattr(ssl, '_create_unverified_context'):
+      sslContext = ssl._create_unverified_context()
+   else:
+      sslContext = None
+
+   return Connect(host=host,
```
Comment (tianhao64) [inline pyVim/connect.py]: version=version is missing.

File: pyVim/connect.py [hunk 9]
```diff
@@ -271,6 +271,36 @@ def Connect(host='localhost', port=443, user='root', pwd='',
...
+                  port=port,
+                  user=user,
+                  pwd=pwd,
+                  service=service,
+                  adapter=adapter,
+                  version=version,
```
Comment (JLHasson) [inline pyVim/connect.py]: The issue is here?

File: pyVim/connect.py (hunk 8 above)
Comment (tianhao64) [inline pyVim/connect.py]: namespace=namespace is missing

**Pull request merged**